src/Compilers/CSharp/Portable/Binder/Semantics/OverloadResolution/MethodTypeInference.cs

```diff
@@ -2733,11 +2733,13 @@ private bool Fix(int iParam, ref CompoundUseSiteInfo<AssemblySymbol> useSiteInfo
             if (containsFunctionTypes(lower) &&
                 (containsNonFunctionTypes(lower) || containsNonFunctionTypes(exact) || containsNonFunctionTypes(upper)))
             {
-                lower = removeTypes(lower, static type => isFunctionType(type, out _));
+                removeTypes(lower, static type => isFunctionType(type, out _));
+            }
+            else
+            {
+                // Remove any function types with no delegate type.
+                removeTypes(lower, static type => isFunctionType(type, out var functionType) && functionType.GetInternalDelegateType() is null);
```

Comment:
I moved this to `else`, because if the first condition is true, this won't do anything.
This is removing types with a stricter condition.

If we already removed types where `isFunctionType(type)` is true. Then there is no point to try to remove types where `isFunctionType(type) && anything` is true.

Reply:
This change moved to #69070